### ISSUE TITLE
Add a CLI-only app

### ISSUE THREAD

**Issue by BlueDrink9** [autokey/autokey#342]
Once the app refactor of #341 is done, I would like to create a CLI-only app, which runs all necessary app and engine setup, but without a GUI.
Even if the only thing it does is run a functional dbus server, once the scripting API is fully featured, a user could run the CLI app then use autokey-run to perform script actions.

Alternatively, the CLI-app could run scripts directly on startup, so users could specify `autokey-cli autokeyrc.py` and have everything they need set up. Combined with only making `temporary` items, they could then `C-c`, edit the script, then run again to update their config (as nix users are used to doing with their other tools).

**Comment by josephj11** [comment]
I'm a bit confused by this issue. autokey-run already exists for running scripts outside of AutoKey, but it depends on AutoKey running at the same time. Are you trying to make external scripts run without having the normal AutoKey running?

Do you want to have AutoKey do things for you, but not have the user interactive level (hotkeys and trigger phrases) available to the user?

-----

I have a bash script which sets up my KDE session when it starts (using KDE's autostart feature and xdotool). Things like this can be tricky to get right because of timing issues. KDE runs my script before it's done setting up all its windows, etc., so my script has to delay doing anything for awhile because the things it wants to manipulate aren't there yet. And, when they do get there, things like Thunderbird and Vivaldi can be take their time setting up some of their "windows" before others while also creating a bunch of non-visible "windows" which are not user oriented.

I bring this up because, now when you run an AutoKey event, it already has a live/user expected window to act upon, but when you run something at startup, that's not necessarily the case.

**Comment by BlueDrink9** [comment]
Normal autokey, no GUI.

Hotkeys, abbreviations etc work.

I've updated my description.

**Comment by josephj11** [comment]
OK. So this would allow AutoKey to do its thing, but the user can't change/reconfigure anything because there's no GUI? Sounds like something useful for deployment to end user workstations.

What use case do you have in mind?

How does this affect the context menu of the AutoKey icon in the taskbar? At a minimum, it has to disable or remove the Show Main Window option.

**Comment by BlueDrink9** [comment]
The issue I linked to includes improving the api to allow removing things, so this could be controlled via the scripting api.

You bring it in line with autohotkey, sxhkd and every other similar tool which don't require GUI dependencies.

Some people just don't want to have to deal with GUIs. They're usually buggier because they're harder to maintain. Linux is full of terminal junkies that will never touch something that you can't configure without a GUI.

Thanks for pointing out the change to the context menu.

**Comment by dogweather** [comment]
Yep, I imagine that splitting the software out into separate repos / projects for backend and frontends would be great.
